app/src/main/java/org/schabi/newpipe/ReCaptchaActivity.java

```diff
@@ -155,6 +150,16 @@ public boolean onOptionsItemSelected(final MenuItem item) {
         }
     }
 
+    private String getUrlFromIntent() {
```

Comment:
The method should be renamed.
For me a getter should only get a value and not modify it in between. If you only read the line above `final String url = getUrlFromIntent();` and not the implementation itself, it could cause confusion down the line

Reply:
Ok, I will change the signature and add a test for it

Reply:
Something like `resolveUrlFromIntent` would be ok or any other verb that fits